packages/core/src/jwt.ts

```diff
@@ -1,12 +1,63 @@
 import { decodeBase64 } from './base64';
 
+/**
+ * Recognized JWT Claims Set members, any other members may also be present.
+ * @see {@link https://github.com/panva/jose/blob/main/src/types.d.ts#L532}
+ */
+export interface JWTPayload {
```

Comment:
We could add a generic here that takes an object and `&`s the base type and the generic type, so that we can properly type other claims and also prevent unknown props in certain scenarios